[Bug] show tables fail to list tables for paimon catalog

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)

mysql> CREATE EXTERNAL CATALOG paimon_catalog PROPERTIES (     "type" = "paimon",     "paimon.catalog.type" = "hive",     "hive.metastore.uris" = "thrift://127.0.0.1:9083",     "paimon.warehouse" = "file:///opensource/deploystarrocks/paimon_hive/warehouse" );
Query OK, 0 rows affected (0.06 sec)

mysql> set catalog paimon_catalog;
Query OK, 0 rows affected (0.01 sec)

mysql> use sample_db;
Database changed
mysql> show tables;
ERROR 1064 (HY000): Failed to list views in database sample_db


2026-03-01 12:52:40.643+05:30 ERROR (starrocks-mysql-nio-pool-0|188) [MetadataMgr.listTableNames():283] Failed to listTableNames on [paimon_catalog.sample_db]
java.lang.RuntimeException: Failed to list views in database sample_db
        at org.apache.paimon.hive.HiveCatalog.listViews(HiveCatalog.java:836) ~[paimon-bundle-1.3.1.jar:1.3.1]
        at org.apache.paimon.catalog.DelegateCatalog.listViews(DelegateCatalog.java:322) ~[paimon-bundle-1.3.1.jar:1.3.1]
        at com.starrocks.connector.paimon.PaimonMetadata.listTableNames(PaimonMetadata.java:162) ~[fe-core-main.jar:?]
        at com.starrocks.connector.CatalogConnectorMetadata.listTableNames(CatalogConnectorMetadata.java:131) ~[fe-core-main.jar:?]
        at com.starrocks.server.MetadataMgr.listTableNames(MetadataMgr.java:281) ~[fe-core-main.jar:?]
        at com.starrocks.qe.ShowExecutor$ShowExecutorVisitor.visitShowTableStatement(ShowExecutor.java:615) ~[fe-core-main.jar:?]
        at com.starrocks.qe.ShowExecutor$ShowExecutorVisitor.visitShowTableStatement(ShowExecutor.java:402) ~[fe-core-main.jar:?]
        at com.starrocks.sql.ast.ShowTableStmt.accept(ShowTableStmt.java:77) ~[fe-parser-main.jar:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:94) ~[fe-parser-main.jar:?]
        at com.starrocks.qe.ShowExecutor.execute(ShowExecutor.java:326) ~[fe-core-main.jar:?]
        at com.starrocks.qe.StmtExecutor.handleShow(StmtExecutor.java:2521) ~[fe-core-main.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:1072) ~[fe-core-main.jar:?]
        at com.starrocks.qe.ConnectProcessor.executeQueryAttempt(ConnectProcessor.java:556) ~[fe-core-main.jar:?]
        at com.starrocks.qe.ConnectProcessor.runWithParserStageRetry(ConnectProcessor.java:450) ~[fe-core-main.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:387) ~[fe-core-main.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:767) ~[fe-core-main.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:1149) ~[fe-core-main.jar:?]
        at com.starrocks.mysql.nio.MySQLReadListener.handleRequest(MySQLReadListener.java:152) ~[fe-core-main.jar:?]
        at com.starrocks.mysql.nio.MySQLReadListener.lambda$handleEvent$0(MySQLReadListener.java:92) ~[fe-core-main.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
        at java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: org.apache.thrift.TException: method not implemented
        at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.getTables(HiveMetaStoreClient.java:898) ~[fe-core-main.jar:?]
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
        at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.invoke(RetryingMetaStoreClient.java:212) ~[hive-apache-3.1.2-22.jar:?]
        at jdk.proxy2.$Proxy42.getTables(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
        at org.apache.hadoop.hive.metastore.HiveMetaStoreClient$SynchronizedHandler.invoke(HiveMetaStoreClient.java:816) ~[fe-core-main.jar:?]
        at jdk.proxy2.$Proxy43.getTables(Unknown Source) ~[?:?]
        at org.apache.paimon.hive.HiveCatalog.lambda$listViews$21(HiveCatalog.java:834) ~[paimon-bundle-1.3.1.jar:1.3.1]
        at org.apache.paimon.client.ClientPool$ClientPoolImpl.run(ClientPool.java:68) ~[paimon-bundle-1.3.1.jar:1.3.1]
        at org.apache.paimon.hive.pool.CachedClientPool.run(CachedClientPool.java:135) ~[paimon-bundle-1.3.1.jar:1.3.1]
        at org.apache.paimon.hive.HiveCatalog.listViews(HiveCatalog.java:833) ~[paimon-bundle-1.3.1.jar:1.3.1]
        ... 21 more



### Expected behavior (Required)
show tables should be be success

### Real behavior (Required)

### StarRocks version (Required)
 mysql> select current_version();
+-------------------+
| current_version() |
+-------------------+
| main-40eca82      |
+-------------------+
1 row in set (0.16 sec)

--- a/fe/fe-core/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java
+++ b/fe/fe-core/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java
@@ -895,7 +895,7 @@ public class HiveMetaStoreClient implements IMetaStoreClient, AutoCloseable {
     @Override
     public List<String> getTables(String dbName, String tablePattern, TableType tableType)
             throws MetaException, TException, UnknownDBException {
-        throw new TException("method not implemented");
+        return client.get_all_tables(dbName);
     }
 
     @Override

